aquadoggo/src/materializer/service.rs

```diff
@@ -213,8 +227,11 @@ mod tests {
             tx.send(crate::bus::ServiceMessage::NewOperation(first_operation_id))
                 .unwrap();
 
-            // Wait a little bit for work being done ..
-            tokio::time::sleep(Duration::from_millis(100)).await;
+            // Wait for the document to be materialised.
+            while !matches!(
+                rx_status.recv().await.unwrap(),
+                ServiceStatusMessage::Materialiser(TaskStatus::Completed(_))
+            ) {}
```

Comment:
Wondering of this could be expressed a little differently, maybe with the `select` macro of tokio? Or we make it an own function?

Reply:
I can'd find a way to do it with `select`, with a `loop` we could do it like:

```
loop {
    if let ServiceStatusMessage::Materialiser(TaskStatus::Completed(_)) =
        rx_status.recv().await.unwrap()
    {
        break;
    }
}

```

Reply:
Did ypu check the chapter "Pattern Matching" here? https://tokio.rs/tokio/tutorial/select

Otherwise we could refactor this too a simpler messaging format, similar to "trigger". The message is currently binary: pending / complete, it could be simply expressed by something which fires or not.

Reply:
This verbose expression only exists because there is a `Pending` type, which is nowhere used, this is why we have to loop over it, otherwise recv would already be enough.

Reply:
We could introduce a new (more compact) message type instead of re-using `TaskStatus` and we could only send messages from the materializer service when a task completes. We don't really need do know about pending tasks here, that's more a debugging concern.

This would in turn simplify the wait/matching patterns required in simple cases.

Reply:
Maybe we don't need to know about dependency tasks at all either...